python/mxnet/rnn/rnn_cell.py

```diff
@@ -431,8 +431,8 @@ def __init__(self, num_hidden, prefix='lstm_', params=None, forget_bias=1.0):
 
     @property
     def state_info(self):
-        return [{'shape': (0, self._num_hidden), '__layout__': 'NC'},
-                {'shape': (0, self._num_hidden), '__layout__': 'NC'}]
+        return [{'shape': (-1, self._num_hidden), '__layout__': 'NC'},
```

Comment:
I will revert all the changes in this file since we have a global switch to control numpy compatibility now.